internal/charmhub/download.go

```diff
@@ -243,36 +207,23 @@ func (c *DownloadClient) download(ctx context.Context, resourceURL *url.URL, arc
 		progressBar = opts.progressBar
 	}
 
-	hasher := c.getHasher(opts.digestType)
+	hasher256 := sha256.New()
+	hasher384 := sha512.New384()
 
-	writer := io.MultiWriter(f, hasher, progressBar)
-	size, err := io.Copy(writer, r.Body)
+	size, err := io.Copy(f, io.TeeReader(r.Body, io.MultiWriter(hasher256, hasher384, progressBar)))
```

Comment:
This could be simplified to `io.Copy(io.MultiWriter(f, hasher256, hasher384, progressBar), r.Body`) right?

Reply:
I did have that originally.